packages/google_maps_flutter/example/test_driver/google_maps_e2e.dart

```diff
@@ -718,6 +718,35 @@ void main() {
     expect(topLeft, const ScreenCoordinate(x: 0, y: 0));
   });
 
+  testWidgets('testGetDistance', (WidgetTester tester) async {
+    final Key key = GlobalKey();
+    final Completer<GoogleMapController> controllerCompleter =
+        Completer<GoogleMapController>();
+
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: GoogleMap(
+        key: key,
+        initialCameraPosition: _kInitialCameraPosition,
+        onMapCreated: (GoogleMapController controller) {
+          controllerCompleter.complete(controller);
+        },
+      ),
+    ));
+    final GoogleMapController controller = await controllerCompleter.future;
+
+    // We suspected a bug in the iOS Google Maps SDK caused the camera is not properly positioned at
+    // initialization. https://github.com/flutter/flutter/issues/24806
+    // This temporary workaround fix is provided while the actual fix in the Google Maps SDK is
+    // still being investigated.
+    // TODO(cyanglaz): Remove this temporary fix once the Maps SDK issue is resolved.
+    // https://github.com/flutter/flutter/issues/27550
+    await tester.pumpAndSettle(const Duration(seconds: 3));
+
+    final double distance = await controller.getDistance(1, _kInitialMapCenter);
+    expect(distance, 0.00020464534463826567);
```

Comment:
Is there a way that we can test this without using some magic number?

Reply:
I assume the math specified in [Google Groups forums](https://groups.google.com/forum/#!msg/google-maps-js-api-v3/hDRO4oHVSeM/osOYQYXg2oUJ)
`meters/px = 156543.03392 * Math.cos(latitude * Math.PI / 180) / Math.pow(2, zoom)
` can be used to test this and **add android implementation**, But using this method, there was a precision mismatch between the value generated from projection api. This magic number specified here is the value we get from projection api for 1 meter at latitude 0 and longitude -0. I will check if I can get the math correct. help here is appreciated.

Reply:
maybe we can try https://api.flutter.dev/flutter/flutter_test/moreOrLessEquals.html

Reply:
```
final double distance = await inspector.getDistance(1, _kInitialCameraPosition.target);
// Formula is from Chris Broadfoot's comment : https://groups.google.com/d/msg/google-maps-js-api-v3/hDRO4oHVSeM/osOYQYXg2oUJ
final double meters_per_pixel = 156543.03392 * cos(_kInitialCameraPosition.target.latitude * pi / 180) / pow(2, _kInitialZoomLevel);
final double points_per_meter = 1 / meters_per_pixel;
expect(distance, moreOrLessEquals(points_per_meter));
```
When I try the above code, I am getting a test failure.
```
flutter: The following TestFailure object was thrown running a test:
flutter:   Expected: 0.00020441663355236448 (±1e-10)
flutter:   Actual: <0.00020464534463826567>
flutter:    Which: 0.00020464534463826567 is not in the range of 0.00020441663355236448 (±1e-10).
```
for different zoom levels, the actual value and expected value starts to diverge from 6th scale.